orttraining/orttraining/training_api/include/onnxruntime_training_cxx_inline.h

```diff
@@ -0,0 +1,91 @@
+  Env env = Env();
+  ThrowOnError(GetTrainingApi().CreateTrainingSession(
+      env, session_options, checkpoint_state,
+      train_model_path.c_str(),
+      eval_model_path.has_value() ? eval_model_path.value().c_str() : nullptr,
+      optimizer_model_path.has_value() ? optimizer_model_path.value().c_str() : nullptr,
+      &p_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetTrainingModelOutputCount(p_, &training_model_output_count_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetEvalModelOutputCount(p_, &eval_model_output_count_));
+}
+
+inline std::vector<Value> TrainingSession::TrainStep(const std::vector<Value>& input_values) {
+  std::vector<Value> output_values;
+  for (size_t i = 0; i < training_model_output_count_; i++) output_values.emplace_back(nullptr);
+  auto ort_input_values = reinterpret_cast<const OrtValue**>(const_cast<Value*>(input_values.data()));
+  auto ort_output_values = reinterpret_cast<OrtValue**>(output_values.data());
+  RunOptions run_options;
+  ThrowOnError(GetTrainingApi().TrainStep(
+      p_, run_options, input_values.size(), ort_input_values,
+      training_model_output_count_, ort_output_values));
+
+  return output_values;
+}
+
+inline void TrainingSession::ResetGrad() {
+  ThrowOnError(GetTrainingApi().ResetGrad(p_));
+}
+
+inline std::vector<Value> TrainingSession::EvalStep(const std::vector<Value>& input_values) {
+  std::vector<Value> output_values;
+  for (size_t i = 0; i < eval_model_output_count_; i++) output_values.emplace_back(nullptr);
+  auto ort_input_values = reinterpret_cast<const OrtValue**>(const_cast<Value*>(input_values.data()));
+  auto ort_output_values = reinterpret_cast<OrtValue**>(output_values.data());
+  RunOptions run_options;
+  ThrowOnError(GetTrainingApi().EvalStep(
+      p_, run_options, input_values.size(), ort_input_values,
+      training_model_output_count_, ort_output_values));
+
+  return output_values;
+}
+
+inline void TrainingSession::SetLearningRate(float learning_rate) {
+  ThrowOnError(GetTrainingApi().SetLearningRate(p_, learning_rate));
+}
+
+inline float TrainingSession::GetLearningRate() {
+  float learning_rate = 0;
+  ThrowOnError(GetTrainingApi().GetLearningRate(p_, &learning_rate));
+  return learning_rate;
+}
+
+inline void TrainingSession::RegisterLinearLRScheduler(const int64_t warmup_step_count, const int64_t total_step_count,
+                                                       const float initial_lr) {
+  ThrowOnError(GetTrainingApi().RegisterLinearLRScheduler(p_, warmup_step_count, total_step_count,
+                                                          initial_lr));
+}
+
+inline void TrainingSession::SchedulerStep() {
+  ThrowOnError(GetTrainingApi().SchedulerStep(p_));
+}
+
+inline void TrainingSession::OptimizerStep() {
+  RunOptions run_options;
+  ThrowOnError(GetTrainingApi().OptimizerStep(p_, run_options));
+}
+
+inline CheckpointState LoadCheckpoint(const std::basic_string<ORTCHAR_T>& path_to_checkpoint) {
```

Comment:
Do we need this? Can we simply include this within the Checkpoint class?

Reply:
Do you mean like a static member method? 

I am not very inclined on using a constructor parameter for the checkpoint path for `CheckpointState`. As this obfuscates the intention of what the path is used for (is it used for loading, vs saving the checkpoint). Having explicit load and save functions seems clearer.

Reply:
Added `SaveCheckpoint` and `LoadCheckpoint` as static member methods.